oneoffs/heatmap.py

```diff
@@ -34,53 +35,65 @@
 
 
 def eval_for_policy(eval_positions, model_dir, data_dir, idx_start, eval_every):
-  model_paths = oneoff_utils.get_model_paths(model_dir)
+    """Evaluate all positions with all models save the policy heatmaps as CSVs
 
-  print("Evaluating models {}-{}, eval_every={}".format(
-        idx_start, len(model_paths), eval_every))
-  for idx in tqdm(range(idx_start, len(model_paths), eval_every)):
-    if idx == idx_start:
-      player = oneoff_utils.load_player(model_paths[idx])
-    else:
-      oneoff_utils.restore_params(model_paths[idx], player)
+    CSV name is "heatmap-<position_name>-<model-index>.csv"
+    CSV format is: model number, value network output, policy network outputs
 
-    pos_names, positions = zip(*eval_positions)
-    eval_probs, eval_values = player.network.run_many(positions)
+    position_name is taken from the SGF file
+    Policy network outputs (19x19) are saved in flat order (see coord.from_flat)
+    """
 
-    for pos_name, probs, value in zip(pos_names, eval_probs, eval_values):
-      save_file = os.path.join(data_dir, "eval-{}-{}".format(pos_name, idx))
+    model_paths = oneoff_utils.get_model_paths(model_dir)
 
-      with open(save_file, "w") as data:
-        data.write("{},  {},  {}\n".format(
-            idx, value, ",".join(map(str, probs))))
+    print("Evaluating models {}-{}, eval_every={}".format(
+          idx_start, len(model_paths), eval_every))
+    for idx in tqdm(range(idx_start, len(model_paths), eval_every)):
+        if idx == idx_start:
+            player = oneoff_utils.load_player(model_paths[idx])
+        else:
+            oneoff_utils.restore_params(model_paths[idx], player)
 
+        pos_names, positions = zip(*eval_positions)
+        # This should be batched at somepoint.
+        eval_probs, eval_values = player.network.run_many(positions)
 
-def positions_from_sgfs(sgf_files, include_empty=True):
-  # sgf_replay doesn't like empty sgf file
-  data = []
-  if include_empty:
-    data.append(("empty", go.Position(komi=7.5)))
-  for sgf in sgf_files:
-    sgf_name = os.path.basename(sgf).replace(".sgf", "")
-    positions, moves, _, _ = oneoff_utils.parse_sgf(sgf)
-    final = positions[-1].play_move(moves[-1])
-    data.append((sgf_name, final))
-  return data
+        for pos_name, probs, value in zip(pos_names, eval_probs, eval_values):
+            save_file = os.path.join(
+                data_dir, "heatmap-{}-{}.csv".format(pos_name, idx))
+
+            with open(save_file, "w") as data:
+                data.write("{},  {},  {}\n".format(
+                    idx, value, ",".join(map(str, probs))))
+
+
+def positions_from_sgfs(sgf_files, include_empty_position=True):
+    data = []
+    if include_empty:
```

Comment:
Nit, no change needed -- but it seems weird that you're putting this here, FWIW.  Seems like a "give me some Position objects from SGFs" is a very odd place to put "oh yeah and throw in a blank board position too" -- that might be better off where you use it rather than adding it as a special case here.  YMMV.